api/api.go

```diff
@@ -56,53 +58,56 @@ func CreateDatasetAPI(host, bindAddr, secretKey string, dataStore store.DataStor
 	}()
 }
 
-func routes(host, secretKey string, router *mux.Router, dataStore store.DataStore, urlBuilder *url.Builder, downloadGenerator DownloadsGenerator, healthCheckTimeout time.Duration) *DatasetAPI {
+func routes(cfg config.Configuration, router *mux.Router, dataStore store.DataStore, urlBuilder *url.Builder, downloadGenerator DownloadsGenerator) *DatasetAPI {
 	api := DatasetAPI{
-		privateAuth:        &auth.Authenticator{SecretKey: secretKey, HeaderName: "internal-token"},
-		dataStore:          dataStore,
-		host:               host,
-		internalToken:      secretKey,
-		router:             router,
-		urlBuilder:         urlBuilder,
-		downloadGenerator:  downloadGenerator,
-		healthCheckTimeout: healthCheckTimeout,
+		privateAuth:          &auth.Authenticator{SecretKey: cfg.SecretKey, HeaderName: "Internal-Token"},
```

Comment:
Could we get away with setting this to nil when the private endpoints aren't enabled? Don't know if it needs to exists for some auth checks that might happen within get handlers, so not a problem if not.

Reply:
Yes we can, I updated the code to do this.